packages/plugin-core/src/telemetry/common/VersionProvider.ts

```diff
@@ -0,0 +1,15 @@
+import * as vscode from "vscode";
+
+export class VersionProvider {
```

Comment:
can we make use of [[../packages/plugin-core/src/versionProvider.ts]] or refactor it?

Reply:
This class isn't necessary, I'll remove it and just inline the code 

This is the better methodology to get version though. It's a bit too much work to refactor right now since the existing Version Provider is static - I'm going to mark it as deprecated.

Reply:
I'm not quite following - this methodology requires extension context - the places that invoke the current static `VersionProvider` don't have access to extension context, so it's not easy to refactor that without more changes.

Reply:
yeah, it will take some refactoring work. we don't have to do all of it but can we make some progress (eg. refactor one of the methods)?

Reply:
I'm going to defer for now, as it's getting out of scope of this change - all 4 references to `VersionProvider.version()` are inside static methods themselves.  Any reworking of those will create circular dependencies.